src/gregoriotex/gregoriotex-write.c

```diff
@@ -4028,6 +4028,33 @@ static void suppress_expansion(FILE *const f, const char *text)
     }
 }
 
+static int first_note_near_clef(const gregorio_score *const score) {
+    gregorio_clef_info clef = gregorio_default_clef;
+    if (score->first_voice_info) {
+        clef = score->first_voice_info->initial_clef;
+        if (!clef.secondary_line && !clef.flatted && score->first_syllable
+                && score->first_syllable->elements) {
+            gregorio_element *element = score->first_syllable->elements[0];
+            if (element && element->type == GRE_ELEMENT) {
+                gregorio_glyph *glyph = element->u.first_glyph;
+                if (glyph && glyph->type == GRE_GLYPH) {
+                    signed char clef_pitch = LOW_LINE_PITCH
+                        + ((clef.line - 1) * 2);
+                    gregorio_note *note = glyph->u.notes.first_note;
+                    if (glyph->u.notes.glyph_type == G_PODATUS && note->next) {
+                        note = note->next;
+                    }
+                    if (!(note->u.note.pitch > clef_pitch - 4
```

Comment:
Indeed. I wonder if something could also be checked for high keys and above line things...